.github/workflows/precommit-PR.yml

```diff
@@ -17,28 +17,28 @@ jobs:
       matrix:
         python-version: ["3.8", "3.9", "3.10"]
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v2
+      uses: actions/setup-python@v4
       with:
         python-version: ${{ matrix.python-version }}
+    - name: Install Poetry
+      uses: snok/install-poetry@v1
     - name: Install dependencies
       run: |
-        python -m pip install --upgrade pip
-        python -m pip install pytest pytest-cov
-        pip install -e .
+        poetry install --no-interaction
     - name: Test with pytest
       run: |
-        pytest --cov dfds_ds_toolbox --cov-report=html:coverage
+        poetry run pytest --cov dfds_ds_toolbox --cov-report=html:coverage
 
   precommit:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Python 3.8
-      uses: actions/setup-python@v2
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v4
       with:
-        python-version: 3.8
+        python-version: "3.10"
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
```

Comment:
Why are we using `pip` in this step? I think we should also pin the version of `pre-commit` here and make sure it corresponds to the one installed with `poetry`. Maybe its simpler to just stick with `poetry` here also? :blush:

Reply:
This is just to save time in running this. 
pre-commit does not need e.g. pandas to be installed to be able to run the checks. So I just wanted a quick install.